### PR TITLE
Waku as a Network post

### PR DESCRIPTION
**Pull request by fryorcraken**
This article aims to synthetize latest research and development that are shaping Waku as a network and provide a glimpse to the future.

I thought about writing the article last week but wasn't sure how useful it'd be. However, after the last scalability call, it seems to be needed.

Reviewers: Please do check veracity of statements in this article. If you feel I am asserting something that is still uncertain let me know. Thx!

### CODE REVIEW

**Review by alrevuelta — CHANGES_REQUESTED** [review]
Really nice post, left some comments.

I don't want to block the PR, but would love to use this to further discuss waku network future, and end up having a more extended post with the views of the whole team if thats fine.

Something that I would like to also cover in this post:
Required resources of waku nodes in terms of bandwitdth. Not for operators but for average end users of waku. Will I be able to run waku subscribed to a few communities in my laptop in Europe? And if I were in Africa? In order to answer this question, imho we have to come up with a max bandwidth that pubsub topics will use. I would say this is very important in the concept of "waku as a network".

**Review by jm-clius — APPROVED** [review]
I think this is a good overview. Thanks for writing it!

File: research/2023-04-02-waku-as-a-network.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+As we noted in a previous [forum post](https://forum.vac.dev/t/waku-payment-models/166/3), Waku's protocol
+incentivization model needs to be flexible to accommodate various business models. Flexibility ensures that projects
+can choose how they want to use Waku based on their specific needs.
+
+### Reversing the Incentivization Question
+
+Traditionally, the question of incentivization revolves around how to incentivize operators to run nodes. We'd like to
+reframe the question and instead ask, "How do we pay for the infrastructure?"
+
+Waku does not intend to offer a free lunch. Ethereum's infrastructure relies on transaction fees, where users pay for
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply research/2023-04-02-waku-as-a-network.md:39]: @alrevuelta good point on precision. Would there be an alternative phrasing that would enable a correct description of Ethereum while keeping it simple as explain Ethereum's compensation scheme is not really part of the goal of the post.

What about:

> Ethereum's infrastructure is supported by transaction fees and inflation, with validators receiving rewards from both sources.

File: research/2023-04-02-waku-as-a-network.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Providers could offer trial or limited free services to Waku users, similar to Slack or Discord. Once a trial is expired or outgrown,
+a community owner could pay for more storage or bandwidth, similar to Slack's model.
+Alternatively, individual users could contribute financially, akin to Discord's Server Boost, or by sharing their own
+resources with their community.
+
+We anticipate witnessing various scenarios across the spectrum: from users sharing resources to users paying for access to the network and everything in between.
+
+## Waku Network: Ethereum or Cosmos?
+
+Another perspective is to consider whether the Waku network will resemble Ethereum or Cosmos. We want Waku to be
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply research/2023-04-02-waku-as-a-network.md:69]: > Then ofc mention that anyone is free to create their own network, just like Ethereum can have forks, but that are totally segregated from the main network.

Yes I hesitated whether to write more about private networks but opted against it for sake of conciseness. Will include for the sake of exactitude :)

**Comment by fryorcraken** [comment]
Will continue tomorrow. One more TODO:

> Something that I would like to also cover in this post:
> Required resources of waku nodes in terms of bandwitdth. Not for operators but for average end users of waku. Will I be > able to run waku subscribed to a few communities in my laptop in Europe? And if I were in Africa? In order to answer this > question, imho we have to come up with a max bandwidth that pubsub topics will use. I would say this is very important in the concept of "waku as a network".

I am thinking about either adding an "open problem" section.

Or just add in the _Scalability and Discovery Protocols_ section that splitting the traffic does not mean tthere is a bandwitdh cap per shard, And hence, at this stage, we are not able to guarantee that someone using 4G or someone in Africa will be able to use the network.
I'll mention the case of the light client protocols.
Also probably good to mention the work we are doing:
- Status COmmunities protocol analysis to confirm efficient usage of Waku
- Kurtosis simulations to measure actual bandwitdh usage
- More sharding to segregate control messages.

I wonder if some QoS can be applied there? Could we separate "critical" control messages from "nice ux" ones?
Will probably leave that as an open questions and still mention it,

@alrevuelta if you have thoughts on these idea, please share. Otherwise will update and request new review.

File: research/2023-04-02-waku-as-a-network.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alrevuelta) [reply research/2023-04-02-waku-as-a-network.md:39]: yep, thats perfect.

**Comment by fryorcraken** [comment]
> @fryorcraken
> 
> > I am thinking about either adding an "open problem" section.
> 
> Yep, thats fine. Lets put there the uncapped bandwidth. I'm actually planning some writeups in this area (not saying we should cap it) but on what to expect with different message sizes, shards, msg per second, etc. Related to [this](https://hackmd.io/IYmyyUIdSEuRZ7WbYd6dng?view) but from a different angle and stating the trilema Bandwidth/MsgRate/MsgSize. (leaving msg delay by now).

See https://github.com/vacp2p/vac.dev/pull/105/commits/3bdfc8ca6ca401aee78b796f0a7ea310593a514b

**Comment by fryorcraken** [comment]
> LGTM :). I added an idea draft for Waku Products / Waku Network in the vac book: [here](https://github.com/vacp2p/vac-book/blob/main/scratch/waku_products_and_waku_network.md?rgh-link-date=2023-04-04T09%3A55%3A46Z)

Great.

Your write up looks good. I like the idea of being more detailed on what is possible.

May I suggest to park it for a month or two. Let's dogfood namedspace topics, have RLN testnet 3 under way or even close to completion and it'll be then a good opportunity to do a more detailed post on what the "heteregenous waku network" looks like "right now".

Happy to help or take over with the write up when we go for it.

**Pull request merged**